pyccel/ast/builtins.py

```diff
@@ -260,7 +260,9 @@ class PythonFloat(PyccelAstNode):
     _attribute_nodes = ('_arg',)
 
     def __new__(cls, arg):
-        if isinstance(arg, LiteralFloat):
+        if isinstance(arg.dtype, NativeBool):
+            return PythonInt(arg)
+        elif isinstance(arg, LiteralFloat):
             return LiteralFloat(arg, precision = cls._precision)
         elif isinstance(arg, LiteralInteger):
             return LiteralFloat(arg.p, precision = cls._precision)
```

Comment:
Can we use `arg.python_value` here, or it does not exist for an integer?

Reply:
Sorry, I see that this will be changed in PR #764. Never mind.

Reply:
We can but I think it is beyond the scope of this PR as it is fixed in #764 anyway